contracts/characters.sol

```diff
@@ -198,8 +215,16 @@ contract Characters is Initializable, ERC721Upgradeable, AccessControlUpgradeabl
 
             tokens.push(Character(xp, level, trait, staminaTimestamp));
             cosmetics.push(CharacterCosmetics(0, RandomUtil.combineSeeds(seed, 1)));
-            _mint(minter, tokenID);
-            emit NewCharacter(tokenID, minter);
+            address receiver = minter;
+            if(minter != address(0) && minter != address(0x000000000000000000000000000000000000dEaD) && !hasRole(NO_OWNED_LIMIT, minter) && balanceOf(minter) >= characterLimit) {
+                receiver = address(garrison);
+                garrison.redirectToGarrison(minter, tokenID);
+                _mint(address(garrison), tokenID);
```

Comment:
Not saying this is wrong but redirecting before the actual minting feels weird. No transfer is being made so this is fine but it's like we are emitting events and modifying state vars on something that needs one more line of code before "existing". Just being picky in case the redirectToGarrison in the future gets updated to actually transfer the mint. You can ignore this one if that will never happen.

Reply:
Yeah I was trying to invoke it after transfer initially, but `onERC721Received` was causing the transaction to fail.
It does this
```
require(
    supportedTokenTypes.contains(_tokenAddress) &&
        allCharactersInGarrison.contains(_id),
    "Token ID not listed"
);
```
and `allCharactersInGarrison` are updated in `redirectToGarrison` function.
The thing is `onERC721Received` has to make use of `id` parameter and only `allCharactersInGarrison.contains(_id),` made sense to me.